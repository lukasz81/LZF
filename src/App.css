/*
  Off-white (Page Background) = #f4f4f4
  White (Widget Background) = #ffffff
  Dark Grey (Text, Secondary Buttons) = #474c55
  Green (Primary Buttons, Widget Title) = #6db420
  Yellow (Tertiary Buttons, Page Border) = #fec01e
*/
@import url("https://fonts.googleapis.com/css?family=Open+Sans");
:root {
  --off-white: #f4f4f4;
  --white: #ffffff;
  --dark-grey: #474c55;
  --green: #6db420;
  --yellow: #fec01e; }

body {
  background-color: var(--off-white);
  margin: 0;
  padding: 0;
  font-family: 'Open Sans',  sans-serif; }

.display--flex {
  display: flex; }

.display--block {
  display: block; }

.align-self--center {
  align-self: center; }

.container {
  position: relative;
  max-width: 1200px;
  margin: 0 auto; }

a {
  text-decoration: none;
  color: var(--dark-grey); }

.text--decoration-none {
  text-decoration: none; }

.text--indent {
  text-indent: -10000000px; }

.text--transform-uppercase {
  text-transform: uppercase; }

.text--transform-capitalize {
  text-transform: capitalize; }

input .error {
  box-shadow: 0 0 20px var(--yellow); }
