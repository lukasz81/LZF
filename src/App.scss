/*
  Off-white (Page Background) = #f4f4f4
  White (Widget Background) = #ffffff
  Dark Grey (Text, Secondary Buttons) = #474c55
  Green (Primary Buttons, Widget Title) = #6db420
  Yellow (Tertiary Buttons, Page Border) = #fec01e
*/

@import url('https://fonts.googleapis.com/css?family=Open+Sans');

:root {
    --off-white: #f4f4f4;
    --white: #ffffff;
    --dark-grey: #474c55;
    --green: #6db420;
    --yellow: #fec01e;
}

body {
  background-color: var(--off-white);
  margin:0;
  padding:0;
  font-family: 'Open Sans',  sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
}

html {
  box-sizing: border-box;
}
*, *:before, *:after {
  box-sizing: inherit;
}

.page-header {
  font-size: 34px;
  margin: 50px 0;
}
.page-paragraph {
  font-size: 24px;
  margin: 10px 0 30px 0;
  color: var(--dark-grey)
}

.display {
  &--flex {
    display: flex;
  };
  &--block {
    display: block;
  }
  &--inline-block {
    display: inline-block;
  }
}

.space-between {
  justify-content: space-between;
}

.align-self {
  &--center {
    align-self: center;
  }
  &--flex-end{
    align-self: flex-end;
  }
  &--flex-start{
    align-self: flex-start;
  }
}

.container {
  position: relative;
  max-width: 1100px;
  margin: 0 auto;
}

a {
  text-decoration: none;
  color: var(--dark-grey);
  &:hover {
    color: black;
  }
}

.text {
  &-weight {
    &--bold {
      font-weight: bold;
    }
  }
  &-align {
    &--center {
      text-align: center;
    }
  }
  &--decoration-none {
    text-decoration: none
  }
  &--indent {
    text-indent: -10000000px;
  }
  &-transform {
    &--uppercase {
      text-transform: uppercase;
    }
    &--capitalize {
      text-transform: capitalize;
    }
  }
}

input.error {
  box-shadow: 0 0 20px var(--yellow);
  background: var(--yellow);
}


