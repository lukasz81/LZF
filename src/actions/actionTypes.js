export const ANIMAL_TYPE = 'ANIMAL_TYPE';
export const SET_DONATION_VALUE = 'SET_DONATION_VALUE';
export const SET_MANUAL_DONATION_VALUE = 'SET_MANUAL_DONATION_VALUE';