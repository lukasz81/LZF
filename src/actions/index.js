import {
    ANIMAL_TYPE,
    SET_DONATION_VALUE,
    SET_TO_REGULAR_DONATION,
    REMEMBER_LAST_DONATION_VALUE
} from './actionTypes';

export const changeAnimalType = animal => ({
    type: ANIMAL_TYPE,
    animal
});
export const setDonationValues = donation => ({
    type: SET_DONATION_VALUE,
    donation
});
export const setToRegularDonations = isRegularDonation => ({
    type: SET_TO_REGULAR_DONATION,
    isRegularDonation
});
export const rememberLastDonationValue = value => ({
    type: REMEMBER_LAST_DONATION_VALUE,
    value
});