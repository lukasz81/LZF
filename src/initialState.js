import {DONATION_VALUES} from "./widget/donationValues";

export const initialState = {
    animal: 'giraffe',
    donation: {
        value: DONATION_VALUES.defaultValue,
        isSetManually: false
    },
    isUserSettingIllegalManualValue: false,
    isRegularDonation: true,
    lastRememberedDonationValue: null,
    submitValues: {}
};