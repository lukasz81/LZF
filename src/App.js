import React, { Component } from 'react';
import Header from './header/Header';
import Widget from './widget/Widget';
import Content from './content/Content';
import Footer from './footer/Footer';
import './App.css';

export class App extends Component {
  render() {
    return (
      <div className="App">
          <Header/>
          <Widget/>
          <Content/>
          <Footer/>
      </div>
    );
  }
}

export default App;
