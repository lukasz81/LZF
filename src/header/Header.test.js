import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ConnectedHeader,{Header} from './Header';
import configureMockStore from 'redux-mock-store';
import App from '../App';
import {initialState} from "../initialState";
import * as actions from "../actions";
import * as actionTypes from "../actions/actionTypes";
import TestRenderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureMockStore();
const mockedStore = mockStore(initialState);
const header = <ConnectedHeader store={mockedStore}/>;

describe('<Header/>', () => {

    it('renders Header component in App', () => {
        const container = shallow(<App/>);
        expect(container.find(ConnectedHeader).length).toEqual(1);
    });

    it('renders Header snapshot correctly', () => {
        const content = TestRenderer
            .create(<Header/>)
            .toJSON();
        expect(content).toMatchSnapshot();
    });

    it('should return "error" string when shouldHaveErrorClassName called', () => {
        const container = shallow(<Header isUserSettingIllegalManualValue={true} submitDonation={() => {}}/>);
        const instance = container.instance();
        const className = instance.buttonClassName();
        expect(className).toEqual('error');
    });

    it('should return "" string when shouldHaveErrorClassName called', () => {
        const container = shallow(<Header isUserSettingIllegalManualValue={false} submitDonation={() => {}}/>);
        const instance = container.instance();
        const className = instance.buttonClassName();
        expect(className).toEqual('');
    });

    it('calls submitDonation when clicking button', () => {
        const handleDonateClick = jest.fn();
        const container = shallow(<Header submitDonation={handleDonateClick}/>);
        container.find('.cta').simulate('click');
        expect(handleDonateClick.mock.calls.length).toBe(1);
    });

    it('should map state to props properly', () => {
        const container = shallow(header);
        const props = container.props();
        expect(props.submitValues).toEqual(initialState);
    });

    it('checks "submitDonation" action on dispatching', () => {
        mockedStore.dispatch(actions.submitDonation(initialState));
        expect(mockedStore.getActions()[0].submitValues).toEqual(initialState);
        expect(mockedStore.getActions()[0].type).toEqual(actionTypes.SUBMIT_DONATION)
    });

});