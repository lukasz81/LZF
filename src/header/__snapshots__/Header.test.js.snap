// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> renders Header snapshot correctly 1`] = `
<header
  className="header display--flex"
>
  <div
    className="container display--flex"
  >
    <h1
      className="logo text--indent align-self--center"
    >
      <a
        className="display--block"
        href="./"
        title="Little Zebra Foundation"
      >
        Little Zebra Foundation
      </a>
    </h1>
    <nav
      className="align-self--center"
    >
      <ul
        className="display--flex text-transform--capitalize"
      >
        <li
          className="align-self--center"
        >
          <a
            href=""
          >
            What we do
          </a>
        </li>
        <li
          className="align-self--center"
        >
          <a
            href=""
          >
            Get Involved
          </a>
        </li>
        <li
          className="align-self--center"
        >
          <a
            href=""
          >
            Shop
          </a>
        </li>
        <li
          className="align-self--center"
        >
          <a
            href=""
          >
            News
          </a>
        </li>
      </ul>
    </nav>
    <form
      className="align-self--center"
    >
      <input
        className="search"
        placeholder="Search here"
        type="search"
      />
    </form>
    <button
      className=" cta text-transform--uppercase align-self--center"
      onClick={[Function]}
    >
      Donate now
    </button>
  </div>
</header>
`;
